forge-gui/res/cardsfolder/upcoming/swashbuckler_extraordinaire.txt

```diff
@@ -0,0 +1,16 @@
+Name:Swashbuckler Extraordinaire
+ManaCost:2 R
+Types:Creature Dragon Rogue Warrior
+PT:2/2
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigTreasure | TriggerDescription$ When CARDNAME enters the battlefield, create a Treasure token. (It's an artifact with "{T}, Sacrifice this artifact: Add one mana of any color.")
+SVar:TrigTreasure:DB$ Token | TokenAmount$ 1 | TokenScript$ c_a_treasure_sac | TokenOwner$ You
+T:Mode$ AttackersDeclared | AttackingPlayer$ You | Execute$ TrigSac | OptionalDecider$ You | TriggerZones$ Battlefield | TriggerDescription$ Whenever you attack, you may sacrifice one or more Treasures. When you do, up to that many target creatures gain double strike until end of turn.
+SVar:TrigSac:DB$ Sacrifice | Defined$ You | Amount$ X | SacValid$ Treasure | SacMessage$ treasure | RememberSacrificed$ True | SubAbility$ TrigImmediateTrig
```

Comment:
The Optional must still be on the effect I think
else you always have to sac all

Reply:
should there be two optional? or just move the optional to the sac part

Reply:
It could change AI behaviour but two prompts would be annoying I think